docs/docsite/rst/dev_guide/testing.rst

```diff
@@ -0,0 +1,200 @@
+feature development, so clearing bugs out of the way is one of the best things you can do.
+
+Even if you're not a developer, helping test pull requests for bug fixes and features is still immensely valuable.
+
+This goes for testing new features as well as testing bug fixes.
+
+In many cases, code should add tests that prove it works but that's not ALWAYS possible and tests are not always comprehensive, especially when a user doesn't have access
+to a wide variety of platforms, or that is using an API or web service.
+
+In these cases, live testing against real equipment can be more valuable than automation that runs against simulated interfaces.
+In any case, things should always be tested manually the first time too.
+
+Thankfully helping test ansible is pretty straightforward, assuming you are already used to how ansible works.
+
+Setup: Checking out a Pull Request
+----------------------------------
+
+You can do this by checking out ansible, making a test branch off the main one, merging a GitHub issue, testing,
+and then commenting on that particular issue on GitHub. Here's how:
+
+.. warning::
+   Testing source code from GitHub pull requests sent to us does have some inherent risk, as the source code
+   sent may have mistakes or malicious code that could have a negative impact on your system. We recommend
+   doing all testing on a virtual machine, whether a cloud instance, or locally.  Some users like Vagrant
+   or Docker for this, but they are optional.  It is also useful to have virtual machines of different Linux or
+   other flavors, since some features (apt vs. yum, for example) are specific to those OS versions.
+
+
+Create a fresh area to work::
+
+
+   git clone https://github.com/ansible/ansible.git ansible-pr-testing
+   cd ansible-pr-testing
+
+Next, find the pull request you'd like to test and make note of the line at the top which describes the source
+and destination repositories. It will look something like this::
+
+   Someuser wants to merge 1 commit into ansible:devel from someuser:feature_branch_name
+
+.. note:: Only test ``ansible:devel``
+   It is important that the PR request target be ansible:devel, as we do not accept pull requests into any other branch.  Dot releases are cherry-picked manually by Ansible staff.
+
+The username and branch at the end are the important parts, which will be turned into git commands as follows::
+
+   git checkout -b testing_PRXXXX devel
+   git pull https://github.com/someuser/ansible.git feature_branch_name
+
+The first command creates and switches to a new branch named testing_PRXXXX, where the XXXX is the actual issue number associated with the pull request (for example, 1234). This branch is based on the devel branch. The second command pulls the new code from the users feature branch into the newly created branch.
```

Comment:
I'm a fan of use inline code syntax highlighting for branch names.

devel --> `devel`